assets/javascripts/kitten/components/grid/adaptable-grid/index.js

```diff
@@ -0,0 +1,106 @@
+import React, { createContext, useState, useEffect, useContext } from 'react'
+import styled, { css } from 'styled-components'
+import PropTypes from 'prop-types'
+import { NUM_COLUMNS, GUTTER } from '../../../constants/grid-config'
+import { ScreenConfig } from '../../../constants/screen-config'
+
+const GridProperties = createContext({})
+
+export const AdaptableGrid = ({
+  children,
+  gutter = GUTTER,
+  colNumber = NUM_COLUMNS,
+  colAlign = 'left',
+  ...other
+}) => {
+  const gridProperties = {
+    colAlign,
+    colNumber: colNumber,
+    gutter: gutter,
+  }
+  return (
+    <StyledGrid gutter={gutter} colAlign={colAlign}>
+      <GridProperties.Provider value={gridProperties}>
+        {children}
+      </GridProperties.Provider>
+    </StyledGrid>
+  )
+}
+
+export const AdaptableGridCol = ({ children, col, offset, ...other }) => {
+  const [styles, setStyles] = useState(null)
+  const { colAlign, colNumber, gutter } = useContext(GridProperties)
+  const marginDirection = colAlign === 'right' ? 'right' : 'left'
+
+  useEffect(() => {
+    const props = { ...other }
+
+    const stylesByMediaQuery = Object.keys(ScreenConfig).map(size => {
+      const mediaQuery = size.toLowerCase()
+      const col = props[`col-${mediaQuery}`]
+      const offset = props[`offset-${mediaQuery}`]
+
+      if (!col && !offset) {
+        return false
+      }
+
+      return css`
+        @media (min-width: ${ScreenConfig[size].min}px) {
+          ${col &&
+            css`
+              width: ${(col * 100) / colNumber}%;
+            `}
+          ${offset > 0 &&
+            css`
+              margin-${marginDirection}: ${(offset * 100) / colNumber}%;
+            `}
+          ${offset === 0 &&
+            css`
+              margin-${marginDirection}: 0;
+            `}
+        }
+      `
+    })
+
+    setStyles(stylesByMediaQuery)
```

Comment:
Pourquoi est-ce qu'on a besoin de faire ça au mount ? Est-ce qu'on ne peut pas le faire par défaut ?

Reply:
J'ai un problème de `Too many re-renders. React limits the number of renders to prevent an infinite loop.` si je n'ai pas de `useEffect`… on regardera ça ensemble ?